src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpSemanticTokensHandler.cs

```diff
@@ -22,34 +21,49 @@ public static IEnumerable<IJsonRpcHandler> Enumerate(RequestHandlers handlers)
                     yield return new OmniSharpSemanticTokensHandler(handler, selector);
         }
 
+        internal static readonly ImmutableDictionary<SemanticHighlightClassification, SemanticTokenType> _coreTokenMap =
+            new Dictionary<SemanticHighlightClassification, SemanticTokenType>()
+            {
+                [SemanticHighlightClassification.Comment] = SemanticTokenType.Comment,
+                [SemanticHighlightClassification.Keyword] = SemanticTokenType.Keyword,
+                [SemanticHighlightClassification.NumericLiteral] = SemanticTokenType.Number,
+                [SemanticHighlightClassification.Operator] = SemanticTokenType.Operator,
+                [SemanticHighlightClassification.StringLiteral] = SemanticTokenType.String,
+                [SemanticHighlightClassification.ClassName] = SemanticTokenType.Class,
+                [SemanticHighlightClassification.StructName] = SemanticTokenType.Struct,
+                [SemanticHighlightClassification.NamespaceName] = SemanticTokenType.Namespace,
+                [SemanticHighlightClassification.EnumName] = SemanticTokenType.Enum,
+                [SemanticHighlightClassification.InterfaceName] = SemanticTokenType.Interface,
+                [SemanticHighlightClassification.TypeParameterName] = SemanticTokenType.TypeParameter,
+                [SemanticHighlightClassification.ParameterName] = SemanticTokenType.Parameter,
+                [SemanticHighlightClassification.LocalName] = SemanticTokenType.Variable,
+                [SemanticHighlightClassification.PropertyName] = SemanticTokenType.Property,
+                [SemanticHighlightClassification.MethodName] = SemanticTokenType.Method,
+                [SemanticHighlightClassification.EnumMemberName] = SemanticTokenType.EnumMember,
+                [SemanticHighlightClassification.EventName] = SemanticTokenType.Event,
+                [SemanticHighlightClassification.PreprocessorKeyword] = SemanticTokenType.Macro,
+                [SemanticHighlightClassification.LabelName] = SemanticTokenType.Label,
+            }.ToImmutableDictionary();
+
         private readonly Mef.IRequestHandler<SemanticHighlightRequest, SemanticHighlightResponse> _definitionHandler;
         private readonly DocumentSelector _documentSelector;
 
+        private static string MakeLSPCompatibleString(string str)
+            => char.ToLower(str[0]) + str.Substring(1);
```

Comment:
```suggestion
            => char.ToLower(str[0]) + str[1..];
```

Reply:
I will also say that I spent a minute trying to figure out if there was a way to do this in one simple line without doing a substring at all, using spans, but I would have to make a local out of `char.ToLower(str[0])` to get that to work.

Reply:
I would have to pull in Range to make your suggestion work for net472. Maybe we can clean this all up one day.